src/main/java/de/tum/in/www1/artemis/service/dto/UserPublicInfoDTO.java

```diff
@@ -0,0 +1,165 @@
+package de.tum.in.www1.artemis.service.dto;
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+
+/**
+ * A DTO representing a user with the minimal information allowed to be seen by other users in a course
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class UserPublicInfoDTO {
+
+    @SuppressWarnings("PMD.ShortVariable")
+    private Long id;
+
+    // we need this to differentiate between users with the same name
+    private String login;
+
+    private String name;
+
+    private String firstName;
+
+    private String lastName;
+
+    private Boolean isInstructor;
```

Comment:
Change to enum instead of using multiple boolean flags

Reply:
Will be done in a follow up